Automattic/woocommerce-payments
# Hide IPP references in Transaction settings when Cash on Delivery is disabled

Fixes #6972 

#### Changes proposed in this Pull Request
* ~Expose `available_payment_gateways` in the response of `settings` endpoint~.  EDIT: Changed the approach, now check if Cash on Display is available when calculating `is_card_present_eligible` in the settings.
* Migrate `ManualCaptureControl` component to TSX. (note: I attempted migration of other components used by this one but I'll defer this migration to a dedicated issue since migrating the data layer qualifies for a dedicated issue).

**Before**

<img width="828" alt="Screenshot 2023-11-06 at 16 30 18" src="https://github.com/Automattic/woocommerce-payments/assets/1553182/cdd520a0-4095-4d15-a10c-6f4589dbaddb">

**After**

<img width="828" alt="Screenshot 2023-11-06 at 16 31 08" src="https://github.com/Automattic/woocommerce-payments/assets/1553182/04f9428f-e629-409f-bfd9-13233b50ec7a">










#### Testing instructions





* You'll need a merchat account based on a IPP enabled country: US, CA, GB.
* As a merchant, ensure that `Cash on Delivery` is enabled in `Payments > Settings > All Payment Methods`.
* Navigate to Payments > Settings > WooPayments.
* Scroll down to the Transactions section and check that the description below `Issue an authorization on checkout, and capture later` is: 
```
Charge must be captured on the order details screen within 7 days of authorization, otherwise the authorization and order will be canceled. The setting is not applied to In-Person Payments (please note that In-Person Payments should be captured within 2 days of authorization).
``` 
* Disable `Cash on Delivery`.
* Navigate to Payments > Settings > WooPayments.  Scroll down to the Transactions section and check that the description below `Issue an authorization on checkout, and capture later` is: 
```
Charge must be captured on the order details screen within 7 days of authorization, otherwise the authorization and order will be canceled.
``` 

-------------------


**Post merge**
